sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -50,17 +50,16 @@ public void onException(Throwable exception) {
 
         // * An operation could fail due to an IOException which indicates a connection reset by the server,
         // * or a channel closes unexpectedly because the server stopped taking requests
-        //
-        // Currently, only ClosedChannelException will raise onConnectionEvent since it is more sure of a signal the server is going down.
-
+        // * or the channel has been shutdown gracefully
         if (exception instanceof IOException) {
-
             if (exception instanceof ClosedChannelException) {
                 this.metrics.recordAddressUpdated(this.onConnectionEvent(RntbdConnectionEvent.READ_EOF, exception));
             } else {
-                if (logger.isDebugEnabled()) {
-                    logger.debug("Will not raise the connection state change event for error", exception);
-                }
+                this.metrics.recordAddressUpdated(this.onConnectionEvent(RntbdConnectionEvent.READ_FAILURE, exception));
```

Comment:
I initially was concerned that some netty timeouts might also come as IOExceptions - but that isn't true (they all implement io.netty.channel.ChannelException - so, I agree - this should be good.

Reply:
yes - they should be different types of exceptions